base/iterators.jl

```diff
@@ -453,7 +458,7 @@ end
 _zip_isdone(::Tuple{}, ::Tuple{}) = (false, ())
 
 IteratorSize(::Type{Zip{Is}}) where {Is<:Tuple} = zip_iteratorsize(ntuple(n -> IteratorSize(fieldtype(Is, n)), _counttuple(Is)::Int)...)
```

Comment:
and_iteratoreltype

Reply:
Not sure what you mean here since this line is about size rather than eltype? I left `zip_iteratorsize` untouched because its semantics seemed specific enough to zip

Reply:
oh, I misread it